forc-pkg/src/pkg.rs

```diff
@@ -85,7 +85,21 @@ pub struct PinnedId(u64);
 
 /// The result of successfully compiling a package.
 #[derive(Debug, Clone)]
-pub struct BuiltPackage {
+pub enum BuiltPackage {
+    BuiltContract(BuiltContract),
+    BuiltNonContract(BuiltPackageDetails),
+}
+
+/// The result of successfully compiling a contract package.
+#[derive(Debug, Clone)]
+pub struct BuiltContract {
+    pub without_tests: BuiltPackageDetails,
+    pub with_tests: Option<BuiltPackageDetails>,
+}
+
+/// The details of a built package.
+#[derive(Debug, Clone)]
```

Comment:
Not really sure if this is the cleanest way to do this. I added this struct to hold common information that we have for contract and non-contract packages. Open to any suggestions 🙌